code.js

```diff
@@ -1,3 +1,20 @@
 function binarySearch(list, element) {
+    let min = 0;
+    let max = list.length - 1;
+    
+    if (list.length=1 && list[0]===element)
+    {return 0;}
+    
+    while (min <= max)
```

Comment:
Hmm, this isn't really binary search. You're going through the list linearly after the initial guess.

Reply:
I'm not sure what you mean, lines 9 through 17 seem to be binary search as I understand it.
First we get the midpoint of the two extremes, line 10,
check to see if that's our target value, lines 10 and 11,
if our target value is lower, our new minimum value is the midpoint plus 1, lines 13 and 14,
else our max is the new value, lines 15 and 16,
then repeat until our guess is correct, or we have searched the whole array, the while condition on line 8.

Did you want this completed recursively?
I similarly attempted this with a slightly different algorithm, where the check to see if the guess was correct when only one element was left (min=max) but didn't perform the initial check, so it didn't pass the test cases, would you prefer I do it that way?